modules/distributor/usage/tracker.go

```diff
@@ -0,0 +1,371 @@
+package usage
+
+import (
+	"math"
+	"math/bits"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/cespare/xxhash/v2"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/prometheus/util/strutil"
+
+	v1 "github.com/grafana/tempo/pkg/tempopb/trace/v1"
+)
+
+const (
+	tenantLabel  = "tenant"
+	trackerLabel = "tracker"
+)
+
+var emptyHash = hash(nil, nil)
+
+type tenantLabelsFunc func(string) []string
+
+type bucket struct {
```

Comment:
Agree, it didn't seem worth pulling it out of the generator, since this only needs counters, but also I wanted to try to tweak the structure a little bit.  It is precomputing the metric description up-front instead of storing the key=value labels.